### PR TITLE
Added tools dnu

### PR DESCRIPTION
**Pull request by jorisdebock**
Added support for DNX Utility

- dnu restore
- dnu build
- dnu pack

### CODE REVIEW

**Comment by patriksvensson** [comment]
@jorisdebock Great! :+1: Will take a look at this tonight (CET).

**Comment by jorisdebock** [comment]
@patriksvensson did you have time to take a look at it?

**Comment by devlead** [comment]
Sorry for not getting back to you @jorisdebock , if no one gets to it before me I'll take a look and get back to latest this weekend.

**Comment by patriksvensson** [comment]
@jorisdebock Sorry for not getting back to this sooner. Life got in the way of reviewing this.

As far as I can see this looks good, but there are some minor things that need to be addressed.

1. All references to `Dnu` in comments (and alias attributes etc) should be `DNU`.
2. All comments should end with a period (`.`).

Other than that, I think everything looks good :+1: 

@devlead @gep13 You have more experience than me with :NET Core, would you mind taking a look as well?

**Comment by jorisdebock** [comment]
@patriksvensson I fixed the minor things you said, let me know if there are other things that can be improved.

**Comment by patriksvensson** [comment]
@jorisdebock Looks good to me :+1: 

Could you rebase this against develop and squash it to a single commit?
Would also appreciate if you updated the commit message to something like `Added DNU tool support.`

Thanks! :smile:

File: src/Cake.Common/Tools/DNU/DNUAliases.cs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        ///         Proxy = "exampleproxy",
+        ///         NoCache = true,
+        ///         Packages = "./packages",
+        ///         IgnoreFailedSources = true,
+        ///         Quiet = true,
+        ///         Parallel = true,
+        ///         Locked = DNULocked.Lock,
+        ///         Runtimes = new[] { "runtime1", "runtime2" }
+        ///     };
+        ///
```
Comment (devlead) [inline src/Cake.Common/Tools/DNU/DNUAliases.cs]: Trailing whitespace

File: src/Cake.Common/Tools/DNU/DNUAliases.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            restorer.Restore(filePath, settings);
+        }
+
+        /// <summary>
+        /// Build all projects.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The projects path.</param>
+        /// <example>
+        /// <code>
```
Comment (devlead) [inline src/Cake.Common/Tools/DNU/DNUAliases.cs]: Trailing white space.

File: src/Cake.Common/Tools/DNU/DNUAliases.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            restorer.Build(path, settings);
+        }
+
+        /// <summary>
+        /// Pack all projects.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="path">The projects path.</param>
+        /// <example>
+        /// <code>
```
Comment (devlead) [inline src/Cake.Common/Tools/DNU/DNUAliases.cs]: Trailing white space.

**Comment by devlead** [comment]
@jorisdebock this looks good to me :+1: 
I just left some minor code style comment.

Just rebase into one commit as [@patriksvensson](https://github.com/cake-build/cake/pull/533#issuecomment-160224004) already commented and we'll get this merged.

**Comment by jorisdebock** [comment]
@devlead ok great, will do shortly

**Comment by devlead** [comment]
Excellent @jorisdebock :+1:

**Comment by patriksvensson** [comment]
@jorisdebock This have been merged. I had to rebase it against develop so it won't show up a merged. Sorry about this but didn't want to bother you once again with a new rebase :smile: 

Great work! It's really appreciated:+1:

**Pull request closed without merging**